lib/parse_cmds.js

```diff
@@ -0,0 +1,543 @@
+// need to set all these locally in function scope
```

Comment:
No, they need to be fixed in you PR. Currently in master they are scoped to each event emitted. In your PR, they will be shared across messages that will almost certainly lead to some nasty race conditions.